packages/realm-react/src/__tests__/cachedObject.test.ts

```diff
@@ -0,0 +1,251 @@
+////////////////////////////////////////////////////////////////////////////
```

Comment:
As discussed in person, it feels like these tests are testing implementation details of the cache rather than the public facing API/results, and so are likely to break as we refactor etc. This functionality should all be being tested by the rendering tests, so let's double check if every case in here is covered (and add a test for any that are missing) and then remove these.

It may be worth adding some basic unit test for the public facing API of `cachedObject`, just checking that it can be constructed etc.? But maybe that's not very useful.